keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NetflixSemVerVersioningStrategy.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.artifacts
+
+import com.netflix.spinnaker.keel.api.artifacts.PublishedArtifact
+import com.netflix.spinnaker.keel.api.artifacts.VersioningStrategy
+import com.netflix.spinnaker.keel.core.NETFLIX_SEMVER_COMPARATOR
+
+/**
+ * A [VersioningStrategy] that codifies the versioning scheme conventions at Netflix.
+ */
+object NetflixSemVerVersioningStrategy : VersioningStrategy {
+  override val type: String = "netflix-semver"
+
+  override val comparator: Comparator<String> =
+    NETFLIX_SEMVER_COMPARATOR
+
+  private val NPM_VERSION_REGEX = Regex("(\\d+\\.\\d+\\.\\d+(-rc(\\.\\d+)?)?)(-[h]?\\w+)?(-\\w+)?")
+  private val BUILD_REGEX = Regex("-[h]?")
+
+  /**
+   * Extracts a version display name from the longer version string, leaving out build and git details if present.
+   */
+  fun getVersionDisplayName(artifact: PublishedArtifact): String {
```

Comment:
I think it's unclear why you have NPM specific stuff in a comparator that's used for both debians and npm artifacts. Can you make this a little cleaner?

Reply:
Oops, yeah, that was a poor choice of a variable name. Should be something like `NETFLIX_VERSION_REGEX`.

Reply:
BTW, I know for a fact that this regex is not matching _all_ the patterns we use, because the "convention" is somewhat loose. I expect to have to revisit the expression later if we find cases that don't match and that we want to support.

Reply:
Fixed.